src/tbbmalloc/backend.cpp

```diff
@@ -317,11 +319,19 @@ inline bool BackendSync::waitTillBlockReleased(intptr_t startModifiedCnt)
         if (currCoalescQInFlyBlocks > 0 && backend->scanCoalescQ(/*forceCoalescQDrop=*/false))
             break;
         // 4) when there are no blocks
-        if (!currBinsInFlyBlocks && !currCoalescQInFlyBlocks)
+        if (!currBinsInFlyBlocks && !currCoalescQInFlyBlocks) {
             // re-scan make sense only if bins were modified since scanned
+            auto pool = backend->extMemPool;
+            if (pool->hardCachesCleanupInProgress || pool->softCachesCleanupInProgress) {
```

Comment:
You should be able to use acquire/release memory order instead of sequential consistency (at least on `*CleanupInProgress`): std::memory_order_acquire for load operations , std::memory_order_release for store operations and std::memory_order_acq_rel for read-modify-write operations (`exchange`)

Reply:
Done